Source/WebCore/platform/graphics/texmap/TextureMapper.cpp

```diff
@@ -1329,14 +1371,50 @@ void TextureMapper::beginClip(const TransformationMatrix& modelViewMatrix, const
     glUseProgram(program->programID());
     glEnableVertexAttribArray(program->vertexLocation());
 
-    unsigned numberOfVertices = polygon.numberOfVertices();
-    Vector<GLfloat> polygonVertices;
-    polygonVertices.reserveCapacity(numberOfVertices * 2);
-    for (unsigned i = 0; i < numberOfVertices; i++) {
-        auto v = polygon.vertexAt(i);
-        polygonVertices.append(v.x());
-        polygonVertices.append(v.y());
-    }
+    // Convert a point to Normalized Device Coordinates (NDC) in the range [-1, 1].
+    static auto posToNDC  = [](const FloatPoint& p, const TransformationMatrix& mvpMatrix) {
+        double x = p.x();
+        double y = p.y();
+        double z = 0;
+        double w = 1;
+        mvpMatrix.map4ComponentPoint(x, y, z, w);
+        x = clampTo(x / w, -1.0, 1.0);
+        y = clampTo(y / w, -1.0, 1.0);
+        return FloatPoint(x, y);
+    };
+
+    // Convert a point from NDC to screen space.
+    static auto ndcToScreen = [](const FloatPoint& ndcPoint, IntRect viewport) {
+        float x = ((ndcPoint.x() + 1.0f) / 2.0f) * viewport.width() + viewport.x();
+        float y = ((ndcPoint.y() + 1.0f) / 2.0f) * viewport.height() + viewport.y();
+        return FloatPoint(x, y);
+    };
+
+    // Compute the scissor rectangle from the clip path bounding box.
+    IntRect viewport(data().viewport[0], data().viewport[1], data().viewport[2], data().viewport[3]);
+    auto mvpMatrix = data().projectionMatrix * modelViewMatrix;
+    auto clipBounds = clipPath.bounds();
+
+    auto clipP1NDC = posToNDC(clipBounds.minXMinYCorner(), mvpMatrix);
+    auto clipP2NDC = posToNDC(clipBounds.maxXMinYCorner(), mvpMatrix);
+    auto clipP3NDC = posToNDC(clipBounds.maxXMaxYCorner(), mvpMatrix);
+    auto clipP4NDC = posToNDC(clipBounds.minXMaxYCorner(), mvpMatrix);
```

Comment:
Why do you need to convert to NDC? You convert the NDC points to screen coord. Why not convert them to screen coord and take min/max?

Reply:
I’ve updated the implementation to use the `mapQuad`